providers/default.rb

```diff
@@ -391,10 +435,10 @@ def execute_install_script(install_script)
           Get-Service chef-client -ErrorAction SilentlyContinue | stop-service
           Get-Service push-jobs-client -ErrorAction SilentlyContinue | stop-service
 
-          if ((Get-WmiObject Win32_Process -Filter "name = 'ruby.exe'" | Select-Object CommandLine | select-string 'opscode').count -gt 0) {
-            Write-Output "Chef cannot be upgraded while in use. Exiting..."
-            exit 8
-          }
+          #{reschedule_task_function}
+          #{wait_for_chef_client_or_reschedule_upgrade_task_function}
+
+          WaitForChefClientOrRescheduleUpgradeTask
```

Comment:
Add properties that allow users to pass custom values to this function for the `$maxcount` and `$inMinutes`